File: 设计模式/设计模式/命令模式/单品/DanPinSwitchModeCommand.m
//
//  DanPinSwitchModeCommand.m
//  设计模式
//
//  Created by zachary spark on 2019/5/13.
//  Copyright © 2019 3g. All rights reserved.
//

#import "DanPinSwitchModeCommand.h"
@interface DanPinSwitchModeCommand()
@property(nonatomic,strong)DanPinRealization *DanPin;
@end
@implementation DanPinSwitchModeCommand

- (instancetype)init:(DanPinRealization *)DanPin{
    
    self = [super init];
    if(self){
        self.DanPin = DanPin;
    }
    return self;
    
}

- (void)execute{
    [self.DanPin ProductDetectionMode:self.Mode];
}

@end

File: 设计模式-Swift/设计模式-Swift/命令模式/基本实现/DanPinSwitchModeCommand.swift
//
//  DanPinSwitchModeCommand.swift
//  设计模式-Swift
//
//  Created by zachary spark on 2019/5/14.
//  Copyright © 2019 3g. All rights reserved.
//

import UIKit

class DanPinSwitchModeCommand: DanPinCommandProtocol {

    var DanPin:DanPinRealization
    
    var mode:DanPinDetectionMode = .None
    
    init(_ DanPin:DanPinRealization) {
        self.DanPin = DanPin
    }
    
    func execute() {
        self.DanPin.ProductDetectionMode(self.mode as AnyObject)
    }
    
}
